plugins/Phone_fr.py

```diff
@@ -26,59 +26,89 @@ class Phone_fr(Plugin):
 
     only_for = ["FR"]
 
+    PHONE_TAGS = set((u"contact:fax", u"contact:phone", u"fax", u"phone"))
+
     def init(self, logger):
         Plugin.init(self, logger)
 
         import re
-        self.BadShort = re.compile(r"^(\+33 *)([0-9]{4})$")
-        self.BadInter = re.compile(r"^(\+33 *0)([0-9 ]{8,})$")
-        self.National = re.compile(r"^(0 *)([1-9][0-9 ]{8,})$")
-        self.errors[30921] = { "item": 3092, "level": 2, "tag": ["value", "fix:chair"], "desc": T_(u"Extra \"0\" after international code") }
```

Comment:
Please keep the re.compile in the init. Init() is done once. node() is run for each objects.